From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ben Wiederhake <BenWiederhake.GitHub@gmx.de>
Date: Sat, 8 Apr 2023 20:53:52 +0200
Subject: [PATCH] Disable QDnsLookup entirely

It seems that we used to include the host's system headers, which we no
longer do. Now this feature breaks the package, which is why we have to
disable it in order to build Qt6.
---
 src/network/kernel/qdnslookup_unix.cpp | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/src/network/kernel/qdnslookup_unix.cpp b/src/network/kernel/qdnslookup_unix.cpp
index 75f7c6c440e880b93530d2c38fed34a0c143f885..42aaba2699b3feb8e1624db633a792c3168a087f 100644
--- a/src/network/kernel/qdnslookup_unix.cpp
+++ b/src/network/kernel/qdnslookup_unix.cpp
@@ -13,6 +13,11 @@
 
 #include <sys/types.h>
 #include <netinet/in.h>
+
+#if 0
+// FIXME: arpa/nameser.h is not part of Serenity. So either we just disable the feature wholesale,
+// or use the host headers (e.g. libc6-dev), but these often don't work either.
+
 #include <arpa/nameser.h>
 #if !defined(Q_OS_OPENBSD)
 #  include <arpa/nameser_compat.h>
@@ -385,3 +390,21 @@ void QDnsLookupRunnable::query(const int requestType, const QByteArray &requestN
 #endif /* QT_CONFIG(library) */
 
 QT_END_NAMESPACE
+
+#endif
+
+QT_BEGIN_NAMESPACE
+
+using namespace Qt::StringLiterals;
+
+void QDnsLookupRunnable::query(const int requestType, const QByteArray &requestName, const QHostAddress &nameserver, QDnsLookupReply *reply)
+{
+    Q_UNUSED(requestType);
+    Q_UNUSED(requestName);
+    Q_UNUSED(nameserver);
+    reply->error = QDnsLookup::ResolverError;
+    reply->errorString = tr("Resolver library can't be loaded: No runtime library loading support");
+    return;
+}
+
+QT_END_NAMESPACE
